### PR TITLE
Update AutoUI affectedEntries for actions when selected data updates 

### PR DESCRIPTION
**Pull request by myarmolinsky**
Update AutoUI affectedEntries for actions when selected data updates 

Change-type: patch
Signed-off-by: Matthew Yarmolinsky <matthew-timothy@balena.io>

---
##### Contributor checklist

- [ ] I have regenerated screenshots for any affected components with `npm run generate-snapshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

File: src/extra/AutoUI/index.tsx [hunk 1]
```diff
@@ -129,6 +129,12 @@ export const AutoUI = <T extends AutoUIBaseResource<T>>({
 		ActionData<T> | undefined
 	>();
 
+	React.useEffect(() => {
+		if (actionData) {
```
Comment (JSReds) [inline src/extra/AutoUI/index.tsx:133]:
```suggestion
		if (!!actionData) {
```

**Review by JSReds — COMMENTED** [review]
LGTM, let's test it before releasing it. Other than that you can self certify !

**Comment by thgreasi** [comment]
May I ask for a clarification on why we needed to do that in the first place?
We are also calling `setActionData` in the `onActionTriggered`. Should that also be changed?
Afaict both src/extra/AutoUI/Actions/Create.tsx & src/extra/AutoUI/Actions/Update.tsx call onActionTriggered with `affectedEntries: selected`.
Should we instead do this wherever `setSelected()` is called, instead of relying on a `useEffect` and waiting for the extra re-render to finish?

**Comment by myarmolinsky** [comment]
> * May I ask for a clarification on why we needed to do that in the first place?
>   Is this something like "Make AutoUI continue updating the selected item instances passed to AutoUI actions" ?
> * We are also calling `setActionData` in the `onActionTriggered`. Should that also be changed?
>   Afaict both src/extra/AutoUI/Actions/Create.tsx & src/extra/AutoUI/Actions/Update.tsx already call onActionTriggered with `affectedEntries: selected`.
> * Should we instead do this wherever `setSelected()` is called, instead of relying on a `useEffect` and waiting for the extra re-render to finish? (we are only calling `setSelected()` in 3 places atm).

- Yes, it's "Make AutoUI continue updating the selected item instances passed to AutoUI actions"
- They call it, but that means that the triggered action is applied to the selected *at the time of trigger*, it does not continue updating the selected after triggering
- Not sure, that means that the more `setSelected`s we may end up using in the future, the more we have to call `setActionData`. Where do we draw the line? At how many `setSelected`s do we start using the useEffect? It feels like these two are intertwined so I think it makes sense to make the relationship clear via a useEffect with [selected] in the dependency array. If you feel strongly otherwise, I can make the change

**Comment by myarmolinsky** [comment]
@balena-ci I self certify!

**Pull request merged**